src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health = 100;
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(toBite) {
+    if (toBite instanceof Herbivore) {
+      if (!toBite.hidden && toBite.health > 0) {
```

Comment:
You can simplify the condition by combining the checks into a single if statement as per the task requirements: 'It's enough to check that incoming object inside `bite` method is instance of class `Herbivore` and is not hidden within single `if` statement.'